### ISSUE TITLE
pinpoint: results OOMs for rendering.mobile

### ISSUE THREAD

**Issue by sadrulhc** [catapult-project/catapult#4571]
If the full rendering.mobile benchmark is run, the results page runs out of memory, and the iframe containing the results gets killed. From chrome's task-viewer, it looks like the memory usage gradually keeps increasing, and it gets killed when it reaches ~2GB. Before being killed, the results page is in 'Parsing HTML...' step.

Sample run: https://pinpoint-dot-chromeperf.appspot.com/job/13c35246640000

@nedn @benshayden

**Comment by simonhatch** [comment]
I doubt results2 was ever meant for that scale, @benshayden would have ideas on what might be doable in the short term but I'd follow @nedn suggestion and just run the stories you need. If being able to view enormous results files like this is a highly requested feature, we can prioritize it appropriately in a coming Q.

**Comment by sadrulhc** [comment]
Would it be possible to make it easier to specify tags when triggering pinpoint jobs? e.g. perhaps we could include that in the benchmark name? rendering.mobile.key_silk etc?

**Comment by nedn** [comment]
@dave-2 for tags support in pinpoint

**Comment by perezju** [comment]
Another alternative would be to allow pinpoint users to download the results data (either as histograms.json or some plain csv) to process locally themselves (e.g. on colabs or what not). Agree results.html is not really ideal for exploring such big data sets, so maybe not attempt to display it at all.

**Comment by sadrulhc** [comment]
Here is a rendering.desktop run from end of June: https://pinpoint-dot-chromeperf.appspot.com/job/137927c0a40000 The results load OK.

Here is a rendering.desktop run from mid-August (yesterday): https://pinpoint-dot-chromeperf.appspot.com/job/13c35246640000 The results OOMs.

I don't believe the pagesets have grown significantly enough during this time. @amyqiu Can you confirm?

**Comment by amyqiu** [comment]
Yes, the size of rendering.desktop should not have changed much since end of June.

**Comment by amyqiu** [comment]
I'm not sure if this helps, but I ran rendering.desktop yesterday and the results loaded okay for me:
https://pinpoint-dot-chromeperf.appspot.com/job/128ce47a640000

**Comment by dave-2** [comment]
Filed [issue 874940](https://bugs.chromium.org/p/chromium/issues/detail?id=874940) for tags support.

**Comment by chiniforooshan** [comment]
Doesn't adding new metrics (more data per test) change the size of results.html significantly. If it does, then we have actually added new metrics since the end of June.

**Comment by sadrulhc** [comment]
Oh good point, yes! We did add a few new metrics recently. Perhaps we can sunset some of the thread_times metrics?

**Comment by chiniforooshan** [comment]
I agree; we should start retiring legacy smoothness and thread_times metrics. I'll assign this to myself, for now.

**Comment by simonhatch** [comment]
For reference:

https://pinpoint-dot-chromeperf.appspot.com/job/137927c0a40000 -> 305 mb

https://pinpoint-dot-chromeperf.appspot.com/job/13c35246640000 -> 640 mb
